backend/app/controllers/eventsController/db/onTabChange.js

```diff
@@ -1,48 +1,48 @@
+			const ownerRoomName = `${tab.restaurantId}:owner`;
 
-            io.to(ownerRoomName).emit("tab:open", { tableId: tab.tableId, tabId: tab._id });
-            console.log(`[Events] [Delayed ${propagationDelay/1000}s] New tab open request sent in room: "${ownerRoomName}"`);
-        }, propagationDelay);
-    }
+			io.to(ownerRoomName).emit('tab:open', { tableId: tab.tableId, tabId: tab._id });
+			console.log(`[Events] [Delayed ${propagationDelay/1000}s] New tab open request sent in room: "${ownerRoomName}"`);
+		}, propagationDelay);
+	}
 
-    if (mongoEvent.operationType === "update") {
+	if (mongoEvent.operationType === 'update') {
 
-        // Call waiter
-        if (mongoEvent.updateDescription.updatedFields?.callWaiter === 'called') {
-            console.log('  ');
-            const tab = await Tabs.findById(mongoEvent.documentKey._id)
-                .populate('tableId')
-                .exec();
+		// Call waiter
+		if (mongoEvent.updateDescription.updatedFields?.callWaiter === 'called') {
+			console.log('  ');
+			const tab = await Tabs.findById(mongoEvent.documentKey._id)
+				.populate('tableId')
+				.exec();
 
-            const ownerRoomName = `${tab.restaurantId}:owner`;
+			const ownerRoomName = `${tab.restaurantId}:owner`;
 
-            io.to(ownerRoomName).emit("waiter:notification", { tableNo: tab.tableId.tableNo, tabId: tab._id });
-            console.log(`[Events] New waiter request sent in room: "${ownerRoomName}"`);
-        }
+			io.to(ownerRoomName).emit('waiter:notification', { tableNo: tab.tableId.tableNo, tabId: tab._id });
+			console.log(`[Events] New waiter request sent in room: "${ownerRoomName}"`);
+		}
 
-        // Close tab
-        if (mongoEvent.updateDescription.updatedFields?.status === 'closed') {
-            console.log('  ');
-            const tab = await Tabs.findById(mongoEvent.documentKey._id);
+		// Close tab
+		if (mongoEvent.updateDescription.updatedFields?.status === 'closed') {
+			console.log('  ');
+			const tab = await Tabs.findById(mongoEvent.documentKey._id);
 
-            const ownerRoomName = `${tab.restaurantId}:owner`;
-            const tabRoomName = `${tab.restaurantId}:${tab._id}`;
+			const ownerRoomName = `${tab.restaurantId}:owner`;
+			const tabRoomName = `${tab.restaurantId}:${tab._id}`;
 
-            io.to(ownerRoomName).emit("tab:closed", { tableId: tab.tableId, tabId: tab._id });
-            console.log(`[Events] Tab closed sent in room: "${ownerRoomName}"`);
+			io.to(ownerRoomName).emit('tab:closed', { tableId: tab.tableId, tabId: tab._id });
+			console.log(`[Events] Tab closed sent in room: "${ownerRoomName}"`);
 
-            io.to(tabRoomName).emit("tab:closed", { tabId: tab._id });
-            console.log(`[Events] Tab closed sent in room: "${tabRoomName}"`);
-        }
-    }
+			io.to(tabRoomName).emit('tab:closed', { tabId: tab._id });
+			console.log(`[Events] Tab closed sent in room: "${tabRoomName}"`);
+		}
+	}
 };
\ No newline at end of file
```

Comment:
This code patch is a listener function that reacts to events happening in a Mongo database. The function reacts to two events: new tab creation and existing tab updates.

For new tab creation, the function sends a delayed request to open a new tab in a specific room over socket.io.

For existing tab updates, the function checks whether the callWaiter or status properties have changed to called and closed respectively. For a change in callWaiter, the function emits wait staff notification to the same room the new tab event was emitted earlier. When the status property changes to closed, the function emits an event to both the owner's room and the tab-specific room signaling that the tab has been closed.

The code logic appears solid without any obvious bugs being present. One minor suggestion would be to ensure that the indentation throughout the code is uniform to assist with code readability.